### PR TITLE
Restore the IndexError in pygpu_index.

### PR DESCRIPTION
**Pull request by abergeron**
fix regression reported in https://github.com/Theano/libgpuarray/pull/449/files/e2f1f07accad1561db8177d85f119dfd3bcdfa2a#diff-ff48e8755d3dfca5cabbf877f62ff560

### CODE REVIEW

File: pygpu/gpuarray.pyx [hunk 1]
```diff
@@ -1412,7 +1412,10 @@ cdef GpuArray pygpu_index(GpuArray a, const ssize_t *starts,
                           const ssize_t *stops, const ssize_t *steps):
     cdef GpuArray res
     res = new_GpuArray(type(a), a.context, a.base)
-    array_index(res, a, starts, stops, steps)
+    try:
+        array_index(res, a, starts, stops, steps)
+    except ValueError, e:
```
Comment (nouiz) [inline pygpu/gpuarray.pyx]: Why this is needed? Can you add a test to make sure this don't get broken again?

**Comment by nouiz** [comment]
As this is a regression fix, I'll merge now. But adding the test after would be great.

**Comment by nouiz** [comment]
And probably changing the error at the C level to be of the right type.

**Comment by nouiz** [comment]
Should we run Theano gpu tests once to make sure to not break it again?

**Comment by lamblin** [comment]
Ideally, I think we should include the tests in `theano/gpuarray` (at least) in the automatic buildbot for libgpuarray. This would make sure such issues are detected.

**Comment by abergeron** [comment]
Instead of having the Theano PRs depend on the master of libgpuarray, we should depend on a tag and/or the latest release.

This way a nightly buildbot or some such catch these issues.

**Pull request merged**